sklearn/impute.py

```diff
@@ -917,7 +931,24 @@ def fit_transform(self, X, y=None):
             if self.verbose > 0:
                 print('[IterativeImputer] Ending imputation round '
                       '%d/%d, elapsed time %0.2f'
-                      % (i_rnd + 1, self.n_iter, time() - start_t))
+                      % (i_rnd + 1, self.max_iter, time() - start_t))
+
+            # stop early if difference between consecutive imputations goes up.
+            # if so, back off to previous imputation
+            if not self.sample_posterior:
+                norm_diff = np.linalg.norm(Xt - Xt_previous) \
+                            / np.linalg.norm(Xt)
+                if norm_diff > norm_diff_previous:
+                    self.imputation_sequence_ = \
+                        self.imputation_sequence_[:-len(ordered_idx)]
+                    Xt = np.copy(Xt_previous)
+                    self.n_iter_ = i_rnd
+                    print('[IterativeImputer] Early stopping criterion '
+                          'reached. Using result of round %d' % i_rnd)
+                    break
+                else:
+                    Xt_previous = np.copy(Xt)
```

Comment:
I don't get why we're copying in both the if and the else here. I think maybe this one is needed if Xt keeps being updated, but the other is not.

Reply:
You're right. we can do this in the `if` branch: `Xt = Xt_previous`.